tests/test_execution.py

```diff
@@ -55,21 +52,19 @@ def test_tuples_execution_on_same_node(global_execution_env):
 
 
 @pytest.mark.slow
-def test_federated_learning_workflow(global_execution_env):
+def test_federated_learning_workflow(factory, clients, client, data_envs):
```

Comment:
It seems strange to depend from clients and client.

What about initializing client from clients?
```python
client = clients[0]
```

Reply:
Yes we could do this, or we could just rollback the changes by writing this:
```py
clients = network.clients[0]
```

Actually, we import `clients` because we need it a few lines below, in
```py
    assert set([d.owner for d in datasets]) == set([c.node_id for c in clients])
```